src/array.rs

```diff
@@ -978,28 +973,11 @@ impl<T: Element> PyArray<T, Ix1> {
     ///     assert_eq!(pyarray.readonly().as_slice().unwrap(), &[1, 2, 3, 4, 5]);
     /// });
     /// ```
+    #[inline(always)]
     pub fn from_vec<'py>(py: Python<'py>, vec: Vec<T>) -> &'py Self {
         vec.into_pyarray(py)
     }
 
-    /// Construct one-dimension PyArray from a type which implements
-    /// [`ExactSizeIterator`](https://doc.rust-lang.org/std/iter/trait.ExactSizeIterator.html).
-    ///
-    /// # Example
-    /// ```
-    /// use numpy::PyArray;
-    /// use std::collections::BTreeSet;
-    /// let vec = vec![1, 2, 3, 4, 5];
-    /// pyo3::Python::with_gil(|py| {
-    ///     let pyarray = PyArray::from_exact_iter(py, vec.iter().map(|&x| x));
-    ///     assert_eq!(pyarray.readonly().as_slice().unwrap(), &[1, 2, 3, 4, 5]);
-    /// });
-    /// ```
-    pub fn from_exact_iter(py: Python<'_>, iter: impl ExactSizeIterator<Item = T>) -> &Self {
```

Comment:
My personal preference to give a gentler landing for users would be for `from_exact_iter` to be deprecated and just forward to `from_iter`. Then remove outright one or two breaking releases later.

Reply:
Done, `from_exact_iter` is now deprecated and its implementation gutted.